css-flexbox: convert 7 manual-tests to ref-tests

clopez opened the pull request:
* This converts the 6 `css-flexbox align-content_*.html` tests and `flexbox-order-only-flexitems.html` test to ref-tests.

clopez committed:
css-flexbox: convert 7 manual-tests to ref-tests

* This converts the css-flexbox align-content_*.html tests and
flexbox-order-only-flexitems.html test to ref-tests.

clopez commented:
The new ref-tests pass on firefox, chrome and safari: https://github.com/web-platform-tests/wpt/pull/23170/checks

cbiesinger commented on css/css-flexbox/reference/align-content_center-ref.html:
```diff
@@ -0,0 +1,24 @@
+#test {
+  height: 200px;
+  width: 80px;
+  align-content: center;
```
Please remove this line (from all the ref files)

clopez replied:
sure, will do

cbiesinger commented on css/css-flexbox/reference/align-content_space-around-ref.html:
```diff
@@ -0,0 +1,28 @@
+}
+#spacermiddle {
+  width: 50px;
+  height: 17px;
```
Hmm, this does not seem to work well with subpixel layout, but since you say the tests pass I guess it's fine. But ideally the test should be rewritten so that things divide to an integer.

clopez replied:
Great idea! I would change the height to 210 in this case so numbers are perfectly round.

cbiesinger replied:
Sounds good!

clopez committed:
css-flexbox/align-content_space-around.html change height to 210.

That way the numbers of the ref-test spacer become 20 and 10 and
are perfectly round avoiding possible problems with subpixel layout.

clopez committed:
css-flexbox/reference/align-content_*: remove align-content from ref files.

clopez commented:
Pushed 2 new commits with the requested changes

cbiesinger reviewed (approved):
Thanks, looks good assuming tests still pass.

dholbert commented:
The test `align-content_stretch.html` also doesn't pass with sub-pixel layout, which means it fails in Firefox on some platforms (see https://bugzilla.mozilla.org/show_bug.cgi?id=1634377 - basically we're trying to divide 50px three ways, which produces fractional pixel values)

I'll fix the test there to use 10px-taller-container (210px) as it seems we did with another test here, so that we'll end up dividing 60px three ways, which produces more predictable results. :)

dholbert commented:
Also, unrelated to my last comment -- for future reference, we probably shouldn't be reusing element IDs for multiple elements in the same WPT testcase. For example, in e.g. http://wpt.live/css/css-flexbox/reference/align-content_space-around-ref.html , this markup is somewhat bogus:
```css
#spacer {
  width: 50px;
  height: 20px;
} 
```
```html
...<div id=spacer></div><div id=test02>2</div><div id=spacer></div>
```

I guess it works, but those `id=spacer` divs should really be using `class=spacer` instead (and the CSS should have `.spacer` instead of `#spacer`), if we want to have the same CSS rule matching multiple elements.

clopez commented:
@dholbert You're right. I missed that there were still tests not using round divisions.
Thanks for fixing it at https://github.com/web-platform-tests/wpt/pull/23406 !
